Recipes/Recipes.API/Controllers/TagsController.cs

```diff
@@ -1,96 +1,40 @@
 ﻿using Microsoft.AspNetCore.Mvc;
-using Recipes.DataAccess.Entities;
-using Recipes.DataAccess.Filtering;
-using Recipes.DataAccess.Repositories;
+using Recipes.Business.Services.Interfaces;
 using Recipes.Public;
 
 namespace Recipes.API.Controllers;
 
 [ApiController]
 [Route("tags")]
-public class TagsController(IGenericRepository<TagEntity> tagsRepository) : ControllerBase
+public class TagsController(ITagsService tagsService) : ControllerBase
 {
     [HttpGet]
     [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     public async Task<ActionResult<IEnumerable<Tag>>> GetAllTags([FromQuery] int? skip, [FromQuery] int? top)
     {
-        IEnumerable<TagEntity> tags;
-        try
-        {
-            tags = await tagsRepository.GetAllAsync(top, skip);
-        }
-        catch (Exception)
-        {
-            return StatusCode(StatusCodes.Status500InternalServerError);
-        }
-
-        var responseTags = tags.Select(GetTagFromEntity);
-
-        return Ok(responseTags);
+        return Ok(await tagsService.GetAllTagsAsync(skip, top));
     }
 
     [HttpGet("{tagId:int}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     public ActionResult<Tag> GetTag(int tagId)
     {
-        TagEntity? tag;
-        try
-        {
-            tag = tagsRepository.GetById(tagId);
-        }
-        catch (Exception)
-        {
-            return StatusCode(StatusCodes.Status500InternalServerError);
-        }
-
-        if (tag is null) return NotFound();
-
-        var responseTag = GetTagFromEntity(tag);
-
-        return Ok(responseTag);
+        return Ok(tagsService.GetTag(tagId));
```

Comment:
Could you fill me in on how this would handle non-200OK responses?

Reply:
Throws exception -> Exception is handled in middleware

Reply:
Ah, well as long as it works ig